qa/workunits/fs/misc/general_vxattrs.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+# test setfattr remove, and check values of vxattr
+# after remove for vxattr, where possible.
+
+set -ex
+
+mkdir -p dir
+
+#ceph.dir.pin test, def val -1, reset val -1
+getfattr -n ceph.dir.pin dir | grep 'ceph.dir.pin="-1"'
+setfattr -n ceph.dir.pin dir 2>&1 | grep "setfattr: dir: Invalid argument"
+setfattr -n ceph.dir.pin -v 1 dir
+getfattr -n ceph.dir.pin dir | grep 'ceph.dir.pin="1"'
+setfattr -x ceph.dir.pin dir
+getfattr -n ceph.dir.pin dir | grep 'ceph.dir.pin="-1"'
```

Comment:
@chrisphoffman This fails in test run - https://pulpito.ceph.com/vshankar-2024-05-27_06:46:53-fs-wip-vshankar-testing-20240521.105740-debug-testing-default-smithi/7727679/

(and many similar failures with the branch)

```
024-05-27T09:50:07.132 INFO:tasks.workunit.client.0.smithi044.stderr:+ setfattr -x ceph.dir.pin dir
2024-05-27T09:50:07.139 INFO:tasks.workunit.client.0.smithi044.stderr:+ getfattr -n ceph.dir.pin dir
2024-05-27T09:50:07.139 INFO:tasks.workunit.client.0.smithi044.stderr:+ grep 'ceph.dir.pin="-1"'
2024-05-27T09:50:07.147 DEBUG:teuthology.orchestra.run:got remote process result: 1
2024-05-27T09:50:07.147 INFO:tasks.workunit:Stopping ['fs/misc'] on client.0...
```

The test expect the xattr fetch to return -1 post removal, but that does not happen.

Reply:
I found the issue and pushed the fix. The no-op portion of remove xattr looked at ceph.dir.pin's value from `cur` instead of the projected inode value.

I've also tested my patch here: https://pulpito.ceph.com/choffman-2024-05-29_18:11:53-fs-wip-choffman-62793-test2-distro-default-smithi/